src/ucs/vfs/fuse/vfs_fuse.c

```diff
@@ -92,14 +92,16 @@ static int ucs_vfs_fuse_getattr(const char *path, struct stat *stbuf,
 
 static int ucs_vfs_fuse_open(const char *path, struct fuse_file_info *fi)
 {
-    char *data;
+    ucs_string_buffer_t strb;
+    ucs_status_t status;
 
-    data = ucs_vfs_path_read_file(path);
-    if (data == NULL) {
+    ucs_string_buffer_init(&strb);
+    status = ucs_vfs_path_read_file(path, &strb);
+    if (status != UCS_OK) {
         return -ENOENT;
     }
 
-    fi->fh = (uintptr_t)data;
+    fi->fh = (uintptr_t)ucs_string_buffer_cstr(&strb);
```

Comment:
there is an underlying assumption that we can release string buffer's internal string by calling free()
maybe need to add API like ucs_string_buffer_extract_mem which will be documented to return a ucs_free-able pointer and will also fail/strdup if buffer was not allocated by ucs_malloc (e.g fixed)

Reply:
Agree. I will create another PR with `ucs_string_buffer_extract_mem`. I will move the changes in src/ucs/vfs/fuse/vfs_fuse.c to another PR. To unblock merging other changes and to make this PR a little bit lighter.